icu4j/main/tests/core/src/com/ibm/icu/dev/test/impl/UnitsTest.java

```diff
@@ -0,0 +1,276 @@
+// © 2020 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+
+package com.ibm.icu.dev.test.impl;
+
+import com.ibm.icu.dev.test.TestUtil;
+import com.ibm.icu.impl.Pair;
+import com.ibm.icu.impl.units.ComplexUnitsConverter;
+import com.ibm.icu.impl.units.ConversionRates;
+import com.ibm.icu.impl.units.MeasureUnitImpl;
+import com.ibm.icu.impl.units.UnitConverter;
+import com.ibm.icu.impl.units.UnitsRouter;
+import com.ibm.icu.util.Measure;
+import org.junit.Test;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+
+public class UnitsTest {
+
+    public static boolean compareTwoBigDecimal(BigDecimal expected, BigDecimal actual, BigDecimal delta) {
+        BigDecimal diff =
+                expected.abs().compareTo(BigDecimal.ZERO) < 1 ?
+                        expected.subtract(actual).abs() :
+                        (expected.subtract(actual).divide(expected, MathContext.DECIMAL128)).abs();
+
+        if (diff.compareTo(delta) == -1) return true;
+        return false;
+    }
+
+    @Test
+    public void testComplexUnitConverterSorting() {
+
+        MeasureUnitImpl source = MeasureUnitImpl.forIdentifier("meter");
+        MeasureUnitImpl target = MeasureUnitImpl.forIdentifier("inch-and-foot");
+        ConversionRates conversionRates = new ConversionRates();
+
+        ComplexUnitsConverter complexConverter = new ComplexUnitsConverter(source, target, conversionRates);
+        List<Measure> measures = complexConverter.convert(BigDecimal.valueOf(10.0));
+
+        assertEquals(measures.size(), 2);
+        assertEquals("inch-and-foot unit 0", "foot", measures.get(0).getUnit().getIdentifier());
+        assertEquals("inch-and-foot unit 1", "inch", measures.get(1).getUnit().getIdentifier());
+
+        assertTrue("inch-and-foot value 0", compareTwoBigDecimal(BigDecimal.valueOf(32), BigDecimal.valueOf(measures.get(0).getNumber().doubleValue()), BigDecimal.valueOf(0.0001)));
+        assertTrue("inch-and-foot value 1", compareTwoBigDecimal(BigDecimal.valueOf(9.7008), BigDecimal.valueOf(measures.get(1).getNumber().doubleValue()), BigDecimal.valueOf(0.0001)));
```

Comment:
Given that changes are anyway needed for MeasureUnitTest.java, I think you might as well make the Java equivalent of this fix a part of this PR: https://github.com/icu-units/icu/commit/602e41b080fa5d18ad9491ce299938f094a13b9c - I think it's less work?

Given that Shane has approved up to now, I'm happy/confident approving any further changes made, then you should be able to submit without first waiting for Shane to review the additional changes too.